manim/utils/units.py

```diff
@@ -58,9 +71,16 @@ def __init__(self, value):
         self._converter = lambda val, dim: val / 100 * _size_from_dimension(dim)
 
 
-def accepts_unit(f):
+def accepts_unit(f, dim=None):
     @wraps(f)
-    def wrapper(*args):
-        return f(*[a.value if isinstance(a, Unit) else a for a in args])
+    def wrapper(*args, **kwargs):
+        # TODO figure what to do when dim=None, preferable determine the dim.
+        return f(
+            *[v.convert(dim) if isinstance(v, Unit) else v for v in args],
+            **{
+                k: v.convert(dim) if isinstance(v, Unit) else v
+                for k, v in kwargs.items()
```

Comment:
Comment (non-blocking): Let's leave this for now, though in the future I would like to see how much less efficient we are making the whole codebase by supporting `Unit`... :eyes: